pkgs/games/papermc/default.nix

```diff
@@ -1,39 +1,52 @@
-{ lib, stdenv, fetchurl, bash, jre }:
-
-stdenv.mkDerivation rec {
+{
+  lib,
+  stdenvNoCC,
+  fetchurl,
+  jre,
+  makeBinaryWrapper,
+}:
+let
   pname = "papermc";
-  version = "1.20.1.83";
-
-  jar = let
-    mcVersion = lib.versions.pad 3 version;
-    buildNum = builtins.elemAt (lib.versions.splitVersion version) 3;
-  in fetchurl {
-    url = "https://papermc.io/api/v2/projects/paper/versions/${mcVersion}/builds/${buildNum}/downloads/paper-${mcVersion}-${buildNum}.jar";
-    sha256 = "sha256-HQpc3MOXa1wkXqgm9ciQj04FUIyuupnYiu+2RZ/sXE4=";
-  };
+  version = "1.20.2.234";
+in
+stdenvNoCC.mkDerivation {
```

Comment:
A benefit of `mkDerivation` is that the following scheme ensures consistent overriding.

https://github.com/NixOS/nixpkgs/blob/master/doc/stdenv/stdenv.chapter.md#recursive-attributes-in-mkderivation-mkderivation-recursive-attributes

Basically, you need to put `pname` and `version` in `mkDerivation (finalAttrs: { })`, and use `finalAttrs.version` to refer to them in `src` or other places.